02-standard-library/activityMonitor.ts

```diff
@@ -0,0 +1,41 @@
+import { exec } from "child_process";
+import * as fs from "fs";
+
+const logFile: string = "activityMonitor.log";
+
+function logActivity(): void {
+  let command;
+  if (process.platform === "win32") {
+    command =
+      "powershell \"Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + ' ' + $_.CPU + ' ' + $_.WorkingSet }\"";
+  } else if (process.platform === "linux" || process.platform === "darwin") {
+    command = "ps -A -o %cpu,%mem,comm | sort -nr | head -n 1";
+  }
+
+  exec(command, (err: Error | null, stdout: string, stderr: string) => {
+    if (err) {
+      console.log(err.message);
+      return;
+    }
+
+    let result: string = stdout.trim();
+
+    if ((process.stdout as any).clearLine) {
```

Comment:
I'm not sure which node version you are using but, I'm running the LTS version and there stdout is writeStream which does have a clearLine() method, so I am not sure why you need this "as any" and checking if clearLine exists? One thing: clearLine() takes a number param, so it might be good to add it.

writeStream, documentation:
https://nodejs.org/api/tty.html

Screenshot from my machine:
![image](https://github.com/molek95/nodejs-global-mentoring/assets/39033848/1c8a0a3b-7c93-41b2-b6ea-a7da9f25519f)

Please refactor this so it follows writeStream interface and remove as any.

Reply:
Sorry, I haven't payed enough attention to versions and used an older version (other projects that I'm working on uses v14). Now with LTS everything works as you described!